### ISSUE TITLE
Nutscode in Wikidata -> NUTS confusion

### ISSUE THREAD

**Issue by salgo60** [salgo60/Svenskabadplatser#106]
user: [Popperipopp](https://www.wikidata.org/wiki/Topic:W9lvtx8vanb3ulpt) comment of the usage of Nutscode for Swedish bath water in Wikidata

![image](https://user-images.githubusercontent.com/14206509/119529004-fe4fb700-bd81-11eb-8d51-d7389c6d99fb.png)

**Solution** is that we get WD prop proposal [Eionet bathingWaterIdentifier](https://www.wikidata.org/wiki/Wikidata:Property_proposal/bathingWaterIdentifier) approved its in phase **"Create"**

![image](https://user-images.githubusercontent.com/14206509/119529234-348d3680-bd82-11eb-9821-ca900e74bfe1.png)

* also Hav should refactor its use of fieldname indication the wrong thing

**Comment by salgo60** [comment]
**suggested solution**: Hav should refactor the field name

**Comment by salgo60** [comment]
@TGBMonsen så här galet blir min tolkning att Nutskod hos Hav att det är samma som Nutskod i Wikidata [Property:P605](https://www.wikidata.org/wiki/Property:P605) där vi satt upp massa villkor vad som gäller för en Nutskod i WIkidata.....

![image](https://user-images.githubusercontent.com/14206509/119627569-9abe9b80-be0c-11eb-870a-7f5454348cf2.png)

* inte lätt detta med att koppla ihop olika domäner....

**Comment by salgo60** [comment]
;-) 

Jag ringer och pratar med Hav Mattias på måndag vad och när deras plan är om dom tänker rensa i sin röra ;-) 

Trevlig helg

**Comment by TGBMonsen** [comment]
Tack det låter bra, vill gärna höra planerna! Trevlig helg!

**Comment by soleildeminuit** [comment]
Håller med @salgo60 , få HaV att byta ut nutskod/nutscode mot ett korrekt namn, NUTS är reserverat och kan inte användas till "vad som helst". Och när ni ändå är igång, tipsa om att inte blanda svenska och engelska i ett API.  I'm going nuts on this ;)

**Comment by TGBMonsen** [comment]
Tänker att du tar dessa synpunkter på måndag, Salgo60, och återkoppla till mig :) svårt att tro att de hinner ändra nåt innan jag "måste" publicera. Jag väntar ett litet tag men inte jättelänge... :)

**Comment by salgo60** [comment]
@TGBMonsen jo men nu följer du Havs "[anti pattern](https://en.wikipedia.org/wiki/Anti-pattern)" med en term som har fel namn.... om Hav inser problemet med dåligt namnval på termen och kommer att fixa det under 2021 så kan även du springa åt rätt håll direkt..... gissar att skall det du köra ut detta till 290 kommuner som skall sitta i excel med fältnamn, aldrig skapat Öppen Data eller läst en spec så kan det här vara sista versionen du släpper  :cold_sweat: för det är för bökigt att ändra eller?

* när om vi får fältet [EIONET bathingWaterIdentifier](https://www.wikidata.org/wiki/Wikidata:Property_proposal/bathingWaterIdentifier) godkänt i WIkidata så är denna [refraktorering](https://en.wikipedia.org/wiki/Code_refactoring) lite enklare men jag måste  
  * flytta alla badvatten från [Property:P605](https://www.wikidata.org/wiki/Property:P605) **NUTS** till nya fältet 
  - har en surdeg i alla SPARQL frågor jag skrivit mot gamla fältet
  + allt nytt blir dock perfekt och gammal data blir ok plus mindre med ovänner se https://github.com/salgo60/Svenskabadplatser/issues/106 :-)

**Comment by salgo60** [comment]
> Håller med @salgo60 , få HaV att byta ut nutskod/nutscode mot ett korrekt namn, NUTS är reserverat och kan inte användas till "vad som helst". Och när ni ändå är igång, tipsa om att inte blanda svenska och engelska i ett API. I'm going nuts on this ;)

kollar du på https://github.com/salgo60/Svenskabadplatser/issues/118#issuecomment-850795861 så finns det hur mycket som helst att göra
* bort med textsträngar och
  * _responsible_ borde finnas kontaktuppgifter för eller kunna peka på var dessa finns
  * _excecuted measure_ borde ge tillgång till mätvärden eller rapporten
  * _planned measure_ borde peka på ett ärendesystem så man kan följa ärendet och prenumerera på det
* klassificera badet så borde man kan läsa ut det på flera språk
* inget stöd för olika språk utan det verkar genomgående vara strings

känns som en stor resa som måste göras. Steg 0 är väl som alltid att man inser att man har ett problem :tired_face:

Vi kanske skall skapa ett [12 stegs program för dålig data design](https://anonymaalkoholister.se/tolvstegsprogrammet/) där man får gå runt och säga förlåt till alla som använt API:et.... :pray::pray::pray:

**Comment by soleildeminuit** [comment]
Ja, @salgo60 , det finns hur mycket som helst att göra och det borde definitivt finnas en utbildning i API-design för myndigheter. Nu gör alla olika och även inom samma myndighet kan det finnas många varianter, både patterns och antipatterns.  Att i ett API hitta både contaminationSource (eng. singular) och fororeningskallor (svenska plural) är såklart förvirrande och förhindrar effektivt informationsutbyta + risk för fel. 

@TGBMonsen , som @salgo60 är inne på - tänk stort här :) Det kommer bli svårt att ändra i efterhand och att följa standard och vedertagen praxis lönar sig i längden. Tänk även på andra attribut, som rör information där standarder/praxis finns (och de är fler än man kan tro). Jag har inte sett din specifikation (och du har säkert gjort rätt), men vet att i 9 fall av 10 brukar adress behandlas felaktigt. Även geografiska referenssystem brukar bli fel. Det är oerhört viktigt att dessa specas på korrekt sätt och jag har sett alldeles för många specar där detta blivit tokigt.

**Comment by salgo60** [comment]
Japp hur mycket som helst finns det. Det jag skulle vilja se från Hav är att dom tänker större och försöker hitta element i [dd.eionet.europa.eu](https://dd.eionet.europa.eu/searchelements?filter.defaultAttributes%5B2%5D.name=Keywords&filter.defaultAttributes%5B1%5D.id=4&__fp=f256kJU_nwqfHf1SCEwl6FSz34DYywFLGPC0zxz5qeo9Nc5OFx_4ZjPAiY5zmo8fF_3eqXT3Wr5FM2GshlR2UZAiHodTWGYKheYTtEezxDeBSTzY9CWZMWZ6t5X1DALP-BUi1l4TWq2OFmM5WW5hvw%3D%3D&filter.defaultAttributes%5B2%5D.id=5&sort=IDENTIFIER&dir=asc&filter.defaultAttributes%5B2%5D.shortName=Keyword&filter.defaultAttributes%5B0%5D.shortName=Name&filter.defaultAttributes%5B0%5D.value=&filter.defaultAttributes%5B0%5D.id=1&filter.type=&filter.defaultAttributes%5B1%5D.value=&filter.elementType=nonCommon&search=Search&addAttr=0&_sourcePage=Miusa2Nr_VZK7uPz_vre6_WCua3V2e3CskwZJFq5lkzdCHtoiHu1EQ%3D%3D&filter.pageSize=1000&filter.shortName=&filter.defaultAttributes%5B0%5D.name=Name&filter.defaultAttributes%5B1%5D.name=Definition&filter.defaultAttributes%5B2%5D.value=&filter.defaultAttributes%5B1%5D.shortName=Definition&filter.dataSet=&filter.identifier=) och har samma som dom så att det skalar på Europa nivå. Tycker att the Once Only Initiative se https://github.com/salgo60/Svenskabadplatser/issues/118#issuecomment-850862396 borde inspirera folk att tänka stort

* fördelen med galenskapen i Wikidata är att vi "slåss" på samma "yta" så vi måste hitta egenskaper som fungerar för alla, > 300 språk, alla länder, .... etc. vilket det ibland gör ....

**Comment by soleildeminuit** [comment]
Ja, men det skulle kräva att myndigheterna gör en vidare tolkning av regleringsbreven och ser uppdragen som delar i ett större sammanhang. Jag följer med spänning hur väl du lyckas föra fram den saken ;) Om du lyckas, så är mycket vunnet. [https://sv.wikipedia.org/wiki/Kontrollerat_spr%C3%A5k](url)

**Comment by salgo60** [comment]
Kan vi inte få draghjälp från EU https://github.com/salgo60/Svenskabadplatser/issues/118#issuecomment-850862396 ? 

På gott och ont så känner jag att regleringsbrev/instruktioner etc. är det man pekar på för att ha kvar dåliga designmönster och skapa SILOS.... jag tvivlar på det men Corona och EU kanske kan få folk att tänka större och [Once Only Initiative](https://ec.europa.eu/cefdigital/wiki/display/CEFDIGITAL/Once+Only+Principle) tolkar jag som ett annat sätt att säga ni måste jobba digitalt ihop och säga "samma som".... det känns bara galnare och galnare att kommuner skapar [egna dåliga hittepå koordinater för badplatser](https://github.com/salgo60/Svenskabadplatser/issues?q=label%3A%22invalid+coordinate%22+sort%3Aupdated-desc) som sedan skickas till EU.... 

När inte ens Wikipedia kan lita på producerat data känns det som [botten är nådd](https://www.youtube.com/watch?v=ssg9SpkAwio) :worried::worried::worried: - vi ser [samma sak med museerna](http://minancestry.blogspot.com/2020/03/carl-larsson-who-is-that-sadly.html) men där är det ingen som bryr sig att 100tals miljoner Euro kastas bort....

Mönstret med Öppen Data känns det som att saker utreds och utreds i tystnad och många inte vill ha en förändring, de flesta verkar nöjda med dagens "lösningar"...

**Comment by salgo60** [comment]
@TGBMonsen  se https://github.com/salgo60/Svenskabadplatser/issues/118#issuecomment-851926600 vi vet inget mera vad planen är his Hav i deras API. 

Jag stänger denna grej och fokuserar på att datat i Wikidata blir bra. Igår fick badkartan.se [Property:P9615](https://www.wikidata.org/wiki/Property:P9615) godkänt och hoppas snart ha eget fält i Wikidata för [Eionet bathingWaterIdentifier](https://www.wikidata.org/wiki/Wikidata:Property_proposal/bathingWaterIdentifier)

**Comment by salgo60** [comment]
Efterfrågat tydliga termkataloger i Sverige se dataportal.se [topic/120 eionet-data-dictionary](https://community.dataportal.se/topic/120/eionet-data-dictionary)

![image](https://user-images.githubusercontent.com/14206509/120460747-b04a3d00-c399-11eb-9b60-7be2a9da9c15.png)

![image](https://user-images.githubusercontent.com/14206509/120460993-e7b8e980-c399-11eb-9507-15bec4cf6f56.png)

**Comment by salgo60** [comment]
Uppdaterat WIkidata kvalitets matrix med nya egenskaper [P9616](https://www.wikidata.org/wiki/Property:P9616) och [P9615](https://www.wikidata.org/wiki/Property:P9615)
* [Wikidata:WikiProject_Sweden/Svenska_badplatser/Coverage2](https://www.wikidata.org/wiki/Wikidata:WikiProject_Sweden/Svenska_badplatser/Coverage2)

![image](https://user-images.githubusercontent.com/14206509/120541597-5e7dd300-c3ea-11eb-9093-2a70d49ddc8b.png)

Även lagt in dom på [svenska egenskaper](https://www.wikidata.org/wiki/Template:Sweden_properties)
![image](https://user-images.githubusercontent.com/14206509/120541775-9ab13380-c3ea-11eb-848d-c084e8ef3d6a.png)

* senaste [skapade Wikidata egenskaper](w.wiki/DYQ)
* [Wikidata weekly summary](https://meta.wikimedia.org/wiki/Wikidata/Newsletter/Archive#Wikidata_weekly_summary_#318)

**Comment by salgo60** [comment]
Skickade in en fråga

**Comment by salgo60** [comment]
@TGBMonsen @soleildeminuit  Min förståelse efter att [fått detta brev från EEA](https://github.com/salgo60/EuropeanBathingWater/blob/main/Email/Gmail%20-%20Eionet%20bathingWaterIdentifier%20_-_%20Wikidata_Wikipedia.pdf)
* en badplats är en [WFDProtectedArea](https://dd.eionet.europa.eu/vocabulary/wise/WFDProtectedArea/view)
  * [bathingWaterIdentifier](https://dd.eionet.europa.eu/dataelements/99263) _must be a valid bathing water identifier in the "WFDProtectedArea" registry_
    * kollar vi dom vi just nu har i Wikidata [Notebook](https://github.com/salgo60/Svenskabadplatser/blob/main/Jupyter/Eionet%20Data%20Dictionary.ipynb) 
      * [Ok](https://github.com/salgo60/Svenskabadplatser/blob/main/Jupyter/BathIdentifier_Ok.csv) / [Fel](https://github.com/salgo60/Svenskabadplatser/blob/main/Jupyter/BathIdentifier_Error.csv)    
* jag har ändrat formatterings URL:en i wikidata för P9616 se  https://github.com/salgo60/EuropeanBathingWater/issues/3#issuecomment-858318666

**Nästa steg** 
* är att kolla vilka är skit i Wikidata eller finns hos Hav men inte finns i "WFDProtectedArea" registry
* kolla med EEA om enbart EU bad kan finnas i WFDProtectedArea registry
  

![image](https://user-images.githubusercontent.com/14206509/121490599-77cbe400-c9d5-11eb-8e16-f5b42004f764.png)

![image](https://user-images.githubusercontent.com/14206509/121490918-c5e0e780-c9d5-11eb-8164-c86c4459e806.png)

**Comment by salgo60** [comment]
"Snygg" koppling med [Wikidata:Entity_Explosion](https://www.wikidata.org/wiki/Wikidata:Entity_Explosion)

* [Video](https://youtu.be/Ty3nSRMZGp8) 

Koppling till API,  EIONET Data Dictionaryetc..
![image](https://user-images.githubusercontent.com/14206509/121518533-e0748a00-c9f0-11eb-9de0-14377d5f149b.png)

![image](https://user-images.githubusercontent.com/14206509/121518591-f08c6980-c9f0-11eb-9938-322bfd889fba.png)

![image](https://user-images.githubusercontent.com/14206509/121518679-0c900b00-c9f1-11eb-9257-52134dc93ea2.png)

**Comment by salgo60** [comment]
Inget "Intresse" för dialog hos Hav se [email](https://github.com/salgo60/Svenskabadplatser/blob/main/Email/Gmail%20-%20Vem%20hos%20er%20kan%20svara%20p%C3%A5%20anv%C3%A4ndadet%20av%20badvatten%20identifieraren%20-%20bathingWaterIdentifier%202.pdf) - **stänger denna**

**Comment by salgo60** [comment]
@soleildeminuit @TGBMonsen Nu säger Havsmyndigheten att nutskod fösvunnit kanske dax att uppdaterar [Badplatsspecarna](https://lankadedata.se/spec/badplatser/) om det nu är sant.... 

### API dokumentation havochvatten.se/data-kartor-och-rapporter/data-och-statistik

<img width="655" alt="Image" src="https://github.com/user-attachments/assets/80aa92bd-33b6-487b-a286-141ee94bffd3" />

https://www.havochvatten.se/data-kartor-och-rapporter/data-och-statistik/data-och-apier/api-badplatser-och-badvatten.html

### API koll - negativt fortfarande Nutskod
*[badplatsen.havochvatten.se/badplatsen/api/testlocationprofile/SE0110180000001843](https://badplatsen.havochvatten.se/badplatsen/api/testlocationprofile/SE0110180000001843) - ❗ no change

<img width="521" alt="Image" src="https://github.com/user-attachments/assets/159bbfd2-0610-48f0-8949-43987945e220" />

*[badplatsen.havochvatten.se/badplatsen/api/detail/SE0110180000001843](https://badplatsen.havochvatten.se/badplatsen/api/detail/SE0110180000001843) - ❗ no change

<img width="557" alt="Image" src="https://github.com/user-attachments/assets/8a4f52e6-27e2-4191-b18e-3eb793f5bf87" />



### Havsmyndigheten har nytt UI

* [havbipub.havochvatten.se/analytics](https://havbipub.havochvatten.se/analytics/saw.dll?PortalPages&PortalPath=%2Fshared%2FHemsidan%2FBadplats%2F_portal%2FBadplats)

Export till csv ger udda kolumner med EUBathWaterIdentifier som en html länk

* [Provresultat.csv](https://github.com/user-attachments/files/19621406/Provresultat.csv)
* [Badplats (1).xlsx](https://github.com/user-attachments/files/19621419/Badplats.1.xlsx)

<img width="942" alt="Image" src="https://github.com/user-attachments/assets/7b5a64a5-484c-4d2d-b0a7-0226753c82c7" />

**Comment by salgo60** [comment]
### Fråga dataportalen
* [länk](https://community.dataportal.se/topic/421/badplatser/92?_=1743944675052)

<img width="941" alt="Image" src="https://github.com/user-attachments/assets/a72d5278-bbe5-4572-a3af-bd06a4b8ed4f" />

<img width="979" alt="Image" src="https://github.com/user-attachments/assets/c60a6110-9655-46f3-8f42-fbbf03d50f97" />

**Comment by salgo60** [comment]
Email fråga havochvatten@havochvatten.se vad som sker 

[Gmail - Nutskod i Badvatten API_et.pdf](https://github.com/user-attachments/files/19621616/Gmail.-.Nutskod.i.Badvatten.API_et.pdf)

**Comment by salgo60** [comment]
### "Nutskod" history
galet att man inte kan ha en backlog hos Havsmyndiogheten som man kan prenumerera på nu kommuniceras det spretigt på websidor...
* se email till [Havsmyndigheten 2021 jun 3](https://github.com/salgo60/Svenskabadplatser/issues/106#issuecomment-865225071)
* svar [Stina 2021 jun 21](https://github.com/salgo60/Svenskabadplatser/issues/106#issuecomment-865225071)
* versions kommentar [www.havochvatten.se/data-kartor-och-rapporter/data-och-statistik/data-och-apier/api-badplatser-och-badvatten](https://www.havochvatten.se/data-kartor-och-rapporter/data-och-statistik/data-och-apier/api-badplatser-och-badvatten.html) **2025-03-26 Ändrat namnet på variabeln ”nutskod” med ”id”**

**Comment by salgo60** [comment]
### 4 år sedan denna fråga lyftes

Det måste bli enklare att följa Havsmyndighetens ändringar se [diskussion](https://community.dataportal.se/topic/1031/m%C3%B6nster-f%C3%B6r-att-jobba-ihop) [IA](https://web.archive.org/web/20250407065546/https://community.dataportal.se/topic/1031/m%C3%B6nster-f%C3%B6r-att-jobba-ihop)

<img width="842" alt="Image" src="https://github.com/user-attachments/assets/3a73ba22-8f6d-467e-9298-69524c8756b5" />

<img width="1058" alt="Image" src="https://github.com/user-attachments/assets/178b11df-56f3-4277-8603-e4b1e709d1b7" />

<img width="1023" alt="Image" src="https://github.com/user-attachments/assets/6b3714e9-c520-4b08-82fd-5f5b8e217913" />

**Comment by salgo60** [comment]
<img width="938" alt="Image" src="https://github.com/user-attachments/assets/30027393-170d-48f5-bb1a-7a24a0071664" />
